pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -131,14 +132,28 @@ func MergeInboundPolicies(allowPartialHostnamesMatch bool, original []*InboundTr
 }
 
 // MergeOutboundPolicies merges two slices of *OutboundTrafficPolicies so that there is only one traffic policy for a given set of a hostnames
-func MergeOutboundPolicies(original []*OutboundTrafficPolicy, latest ...*OutboundTrafficPolicy) []*OutboundTrafficPolicy {
+// allowPartialHostnamesMatch is set to true when we intend to merge a traffic policy having host headers with other traffic policies
+func MergeOutboundPolicies(allowPartialHostnamesMatch bool, original []*OutboundTrafficPolicy, latest ...*OutboundTrafficPolicy) []*OutboundTrafficPolicy {
 	for _, l := range latest {
 		foundHostnames := false
 		for _, or := range original {
-			if reflect.DeepEqual(or.Hostnames, l.Hostnames) {
-				foundHostnames = true
-				mergedRoutes := mergeRoutesWeightedClusters(or.Routes, l.Routes)
-				or.Routes = mergedRoutes
+			if !allowPartialHostnamesMatch {
+				// For an traffic target outbound policy the hostnames list should fully intersect
+				// to merge the rules
+				if reflect.DeepEqual(or.Hostnames, l.Hostnames) {
+					foundHostnames = true
+					mergedRoutes := mergeRoutesWeightedClusters(or.Routes, l.Routes)
+					or.Routes = mergedRoutes
+				}
+			} else {
+				// When an outbound traffic policy is being merged with another traffic policy (having host headers) the hostnames is not the entire comprehensive list of kubernetes service names
```

Comment:
```suggestion
				// When an outbound traffic policy is being merged with another traffic policy (having host headers) the hostnames are not the entire comprehensive list of kubernetes service names
```

Reply:
Example needed here too for clarity.

Reply:
same as above, rephrased the comment. Please refer unit tests for examples